### DISCUSSION TITLE
[V5] How to update pagination after using query in hook?

### DISCUSSION THREAD

**Discussion by 3rownz** [adonisjs/core#2482] [Help]
Im trying to create soft delete functionality using hooks... similar to how the example is in the [documentation](https://preview.adonisjs.com/guides/models/hooks), This works fine.

My issue is the pagination used in the controller does not detect any of the query changes used in the `@beforeFetch()` hook and returns the total records (deleted and active). The pagination meta data stays the same using original data, prior to adding beforeFetch hook, even though the hook has been added and the data is different. I did notice the `@beforePaginate()` hook exists, but it doesnt work as expected and there is no documentation about this.

How can I update the pagination?

**Product Model:**
```@beforeFetch()
  public static softDeletesFetch = (query: ModelQueryBuilderContract<typeof Product>) => {
    query.whereNull('deleted_at')
  }
```
**Product Controller:**
```
  public async index({ request }) {

    const publicProductColumns = [
      'id',
      'categoryId',
      'name',
      'summary',
      'productCode',
      'manufacturer',
      'price'
    ]

    const currentPage = request.input('current_page', 1)
    const perPage = request.input('per_page', 20)
    const products = Product.query()

    products
      .select(publicProductColumns)
      //other stuff omited for clarity
 
    return await products.paginate(currentPage, perPage)
  }
```

**Output:**
```
 "meta": {
        "total": 369, //<---This stays same
        "per_page": 20,
        "current_page": 1,
        "last_page": 19,
        "first_page": 1,
        "first_page_url": "/?page=1",
        "last_page_url": "/?page=19",
        "next_page_url": "/?page=2",
        "previous_page_url": null
    },
```

**Comment by SchillingPilot** [comment]
Is the `deleted_at` field a proper date field? Double check it has `null` set when there is no date set.

If those are looking correct, set your debug to true as shown here (https://preview.adonisjs.com/guides/database/debugging-queries). Have a look at the query output in your terminal and we can take it from there.
